codex/blockexchange/engine/engine.nim

```diff
@@ -280,6 +269,20 @@ proc scheduleTasks(b: BlockExcEngine, blocksDelivery: seq[BlockDelivery]) {.asyn
 
           break # do next peer
 
+proc issueCancellations(b: BlockExcEngine, addrs: seq[BlockAddress]) {.async.} =
```

Comment:
I would rename this to something like `cancelBlocks`, because `cancelation` is commonly used in the context of `Future`s and it can get confussing.